File: csharp/prime-factors/PrimeFactors.cs
using System.Collections.Generic;

public static class PrimeFactors
{
    public static IEnumerable<long> Factors(long number)
    {
        var factor = 2;

        while (number > 1)
        {
            while (number % factor != 0)
            {
                factor += factor == 2 ? 1 : 2;
            }

            number /= factor;

            yield return factor;
        }
    }
}

File: r/prime-factors/prime-factors.R
prime_factors <- function(n) {
  factors <- c()
  factor <- 2
  
  while (n > 1) {
    while (n %% factor != 0) {
      factor <- factor + ifelse(factor == 2, 1, 2)
    }

    n <- n / factor
    factors <- append(factors, factor)
  }
  
  factors
}
